[Bug] Find Exercise Issue failed

### Summary

When trying to complete step 1, the Find Exercise Issue action failed.

### How to reproduce

Follow instructions step by step and try to complete the exercise in codespace.

<img width="1521" height="586" alt="Image" src="https://github.com/user-attachments/assets/18c4919f-bd61-4778-b315-07836e06b5e7" />

### Additional context

Here the error log:

2026-04-27T21:34:45.2754091Z Current runner version: '2.333.1'
2026-04-27T21:34:45.2775348Z ##[group]Runner Image Provisioner
2026-04-27T21:34:45.2776100Z Hosted Compute Agent
2026-04-27T21:34:45.2776559Z Version: 20260213.493
2026-04-27T21:34:45.2777114Z Commit: 5c115507f6dd24b8de37d8bbe0bb4509d0cc0fa3
2026-04-27T21:34:45.2777723Z Build Date: 2026-02-13T00:28:41Z
2026-04-27T21:34:45.2778341Z Worker ID: {de04a457-37cb-48bc-8abe-6d430480a9a2}
2026-04-27T21:34:45.2778927Z Azure Region: northcentralus
2026-04-27T21:34:45.2779459Z ##[endgroup]
2026-04-27T21:34:45.2780822Z ##[group]Operating System
2026-04-27T21:34:45.2781313Z Ubuntu
2026-04-27T21:34:45.2781806Z 24.04.4
2026-04-27T21:34:45.2782186Z LTS
2026-04-27T21:34:45.2782619Z ##[endgroup]
2026-04-27T21:34:45.2783025Z ##[group]Runner Image
2026-04-27T21:34:45.2783547Z Image: ubuntu-24.04
2026-04-27T21:34:45.2783981Z Version: 20260413.86.1
2026-04-27T21:34:45.2785077Z Included Software: https://github.com/actions/runner-images/blob/ubuntu24/20260413.86/images/ubuntu/Ubuntu2404-Readme.md
2026-04-27T21:34:45.2786330Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu24%2F20260413.86
2026-04-27T21:34:45.2787111Z ##[endgroup]
2026-04-27T21:34:45.2787956Z ##[group]GITHUB_TOKEN Permissions
2026-04-27T21:34:45.2789598Z Issues: read
2026-04-27T21:34:45.2790142Z Metadata: read
2026-04-27T21:34:45.2790817Z ##[endgroup]
2026-04-27T21:34:45.2792753Z Secret source: Actions
2026-04-27T21:34:45.2793409Z Prepare workflow directory
2026-04-27T21:34:45.3178156Z Prepare all required actions
2026-04-27T21:34:45.3232444Z Getting action download info
2026-04-27T21:34:45.6574384Z Download action repository 'actions/github-script@v7' (SHA:f28e40c7f34bde8b3046d885e986cb6290c5673b)
2026-04-27T21:34:46.1490970Z Uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@refs/tags/v0.7.0 (f7e5b28a894634a98b29a4b5802816df69e8c499)
2026-04-27T21:34:46.1495317Z ##[group] Inputs
2026-04-27T21:34:46.1496100Z   issue-title-text: Exercise
2026-04-27T21:34:46.1496743Z ##[endgroup]
2026-04-27T21:34:46.1497418Z Complete job name: Find Exercise Issue / Find exercise by issue title
2026-04-27T21:34:46.2081875Z ##[group]Run actions/github-script@v7
2026-04-27T21:34:46.2082888Z with:
2026-04-27T21:34:46.2087770Z   script: const response = await github.rest.issues.listForRepo({
  owner: context.repo.owner,
  repo: context.repo.repo,
  state: 'all',
  sort: 'created',    // Sort by creation date
  direction: 'desc',  // Newest first (descending order)
  per_page: 50
});

// Get just the issues, not pull requests
const issues = response.data.filter(item => !item.pull_request);

// Start a collapsible group in the logs
core.startGroup(`Found ${issues.length} open issues in total`);
issues.forEach(issue => {
  core.info(`#${issue.number}: ${issue.html_url}`);
});
core.endGroup();

const issue = issues.find(issue => issue.title.includes(process.env.ISSUE_TITLE_PREFIX));
if (!issue) {
  throw new Error('No exercise issue found');
}
core.info(`Exercise issue found: #${issue.number} (${issue.html_url})`);

core.setOutput('ISSUE_URL', issue.html_url);
core.setOutput('ISSUE_NUMBER', issue.number);

2026-04-27T21:34:46.2093295Z   github-token: ***
2026-04-27T21:34:46.2093810Z   debug: false
2026-04-27T21:34:46.2094352Z   user-agent: actions/github-script
2026-04-27T21:34:46.2094944Z   result-encoding: json
2026-04-27T21:34:46.2095458Z   retries: 0
2026-04-27T21:34:46.2096036Z   retry-exempt-status-codes: 400,401,403,404,422
2026-04-27T21:34:46.2096947Z env:
2026-04-27T21:34:46.2097389Z   ISSUE_TITLE_PREFIX: Exercise
2026-04-27T21:34:46.2097975Z ##[endgroup]
2026-04-27T21:34:46.5749066Z ##[group]Found 0 open issues in total
2026-04-27T21:34:46.5751684Z ##[endgroup]
2026-04-27T21:34:46.5752812Z Error: No exercise issue found
2026-04-27T21:34:46.5754931Z     at eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:36187:16), <anonymous>:24:9)
2026-04-27T21:34:46.5757660Z     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2026-04-27T21:34:46.5760404Z     at async main (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:36285:20)
2026-04-27T21:34:46.5786267Z ##[error]Unhandled error: Error: No exercise issue found
2026-04-27T21:34:46.6420918Z Evaluate and set job outputs
2026-04-27T21:34:46.6429413Z Cleaning up orphan processes
2026-04-27T21:34:46.6662482Z ##[warning]Node.js 20 actions are deprecated. The following actions are running on Node.js 20 and may not work as expected: actions/github-script@v7. Actions will be forced to run with Node.js 24 by default starting June 2nd, 2026. Node.js 20 will be removed from the runner on September 16th, 2026. Please check if updated versions of these actions are available that support Node.js 24. To opt into Node.js 24 now, set the FORCE_JAVASCRIPT_ACTIONS_TO_NODE24=true environment variable on the runner or in your workflow file. Once Node.js 24 becomes the default, you can temporarily opt out by setting ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION=true. For more information see: https://github.blog/changelog/2025-09-19-deprecation-of-node-20-on-github-actions-runners/

--- a/.github/workflows/0-start-exercise.yml
+++ b/.github/workflows/0-start-exercise.yml
@@ -18,7 +18,7 @@ jobs:
     if: |
       !github.event.repository.is_template
     name: Start Exercise
-    uses: skills/exercise-toolkit/.github/workflows/start-exercise.yml@v0.7.0
+    uses: skills/exercise-toolkit/.github/workflows/start-exercise.yml@v0.9.3
     with:
       exercise-title: "GitHub Copilot Code Review"
       intro-message: "Leverage GitHub Copilot code review to promote code quality and catch issues before they reach production."
@@ -40,7 +40,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Create comment - add step content
         uses: GrantBirki/comment@v2.1.1

--- a/.github/workflows/1-step.yml
+++ b/.github/workflows/1-step.yml
@@ -16,7 +16,7 @@ env:
 jobs:
   find_exercise:
     name: Find Exercise Issue
-    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.7.0
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.9.3
 
   check_step_work:
     name: Check step work
@@ -35,7 +35,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Find last comment
         id: find-last-comment
@@ -107,7 +107,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Create comment - step finished
         uses: GrantBirki/comment@v2.1.1

--- a/.github/workflows/2-step.yml
+++ b/.github/workflows/2-step.yml
@@ -16,7 +16,7 @@ env:
 jobs:
   find_exercise:
     name: Find Exercise Issue
-    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.7.0
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.9.3
 
   check_step_work:
     name: Check step work
@@ -35,7 +35,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Find last comment
         id: find-last-comment
@@ -102,7 +102,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Create comment - step finished
         uses: GrantBirki/comment@v2.1.1

--- a/.github/workflows/3-step.yml
+++ b/.github/workflows/3-step.yml
@@ -16,7 +16,7 @@ env:
 jobs:
   find_exercise:
     name: Find Exercise Issue
-    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.7.0
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.9.3
 
   check_step_work:
     name: Check step work
@@ -35,7 +35,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Find last comment
         id: find-last-comment
@@ -58,7 +58,7 @@ jobs:
       - name: Check if copilot-instructions.md file exists
         id: check-copilot-instructions
         continue-on-error: true
-        uses: skills/exercise-toolkit/actions/file-exists@v0.7.0
+        uses: skills/exercise-toolkit/actions/file-exists@v0.9.3
         with:
           file: .github/copilot-instructions.md
 
@@ -66,7 +66,7 @@ jobs:
       - name: Check if frontend instructions file exists
         id: check-frontend-instructions
         continue-on-error: true
-        uses: skills/exercise-toolkit/actions/file-exists@v0.7.0
+        uses: skills/exercise-toolkit/actions/file-exists@v0.9.3
         with:
           file: .github/instructions/frontend.instructions.md
 
@@ -74,7 +74,7 @@ jobs:
       - name: Check if backend instructions file exists
         id: check-backend-instructions
         continue-on-error: true
-        uses: skills/exercise-toolkit/actions/file-exists@v0.7.0
+        uses: skills/exercise-toolkit/actions/file-exists@v0.9.3
         with:
           file: .github/instructions/backend.instructions.md
 
@@ -117,7 +117,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Create comment - step finished
         uses: GrantBirki/comment@v2.1.1

--- a/.github/workflows/4-step.yml
+++ b/.github/workflows/4-step.yml
@@ -17,7 +17,7 @@ jobs:
   find_exercise:
     if: github.event.pull_request.merged == true
     name: Find Exercise Issue
-    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.7.0
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.9.3
 
   check_step_work:
     name: Check step work
@@ -36,7 +36,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Find last comment
         id: find-last-comment
@@ -108,7 +108,7 @@ jobs:
         with:
           repository: skills/exercise-toolkit
           path: exercise-toolkit
-          ref: v0.7.0
+          ref: v0.9.3
 
       - name: Create comment - step finished - final review next
         uses: GrantBirki/comment@v2.1.1

